sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparison.scala

```diff
@@ -184,7 +185,7 @@ object UnwrapCastInBinaryComparison extends Rule[LogicalPlan] {
     } else {
       // This means `value` is within range `(min, max)`. Optimize this by moving the cast to the
       // literal side.
-      val lit = Cast(Literal(value), fromType)
+      val lit = Cast(Literal(value), fromType, tz)
```

Comment:
We can also evaluate this and pass the value. It will be optimized away by ConstantFolding anyway.